### PR TITLE
RMSD: added keyword superposition

### LINKED ISSUE #25: rms.RMSD should be able to superimpose
### Expected behaviour
The MDA class superimposes by default. The paralllel `pmda.rms.RMSD` does not superimpose. I would expect the two to behave broadly similar, i.e., the default should be superimposing.

### Actual behaviour
Just raw rmsd calculation. No option to switch on superposition.

### Code to reproduce the behaviour
See #24

### PR DESCRIPTION
**Pull request by orbeckstASU**
FIxes #25 

Changes made in this Pull Request:
- RMSD is now doing a RMSD-superposition by default
- disable with superposition=False

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by orbeckst** [comment]
@kain88-de any idea why the tests died? – It appears that we fail to properly start `distributed.Client()` https://travis-ci.org/MDAnalysis/pmda/jobs/359726635#L2039

Minor thing: How to tell `pytest` to only gather the `test_*.py` files and not also the normal code? https://travis-ci.org/MDAnalysis/pmda/jobs/359726635#L1994 – or is it doing something else??

**Comment by kain88-de** [comment]
> Minor thing: How to tell pytest to only gather the test_*.py files

What you are seeing is the PEP8 test. They only check the formatting of the files. It's best to enforce good code style from the beginning. 

I don't know where the timeout came from. Do the tests pass locally?

**Comment by orbeckst** [comment]
The tests pass locally.

Will check PEP8, thanks.

**Comment by orbeckst** [comment]
The line https://travis-ci.org/MDAnalysis/pmda/jobs/359726637#L2079

> self = LocalCluster('tcp://127.0.0.1:44955', workers=0, ncores=0)

looks suspicious to me: does 0 workers and 0 cores mean "take all of them"? Is there a way to just get two workers with 1 process each with the simple setup or do we need to explicitly start scheduler and workers external of the tests?

**Comment by orbeckst** [comment]
Also, the PEP8 (?) tests say
> pmda/rms.py FAILED 

but I don't see any indication what is wrong. The [pylint job](https://travis-ci.org/MDAnalysis/pmda/jobs/359726638) looks fine.

**Comment by kain88-de** [comment]
> Also, the PEP8 (?) tests say
> 
>     pmda/rms.py FAILED

last line of the log. Pylint has nothing to do with the pep8 check.

**Comment by kain88-de** [comment]
The failure happens because of dask update. If I rerun a master branch build that succeeded I get the same error you have.  Unfortunately I don't have time into looking for a proper fix.

**Comment by orbeckstASU** [comment]
Thanks for all the comments!

--
Oliver Beckstein
Assistant Professor of Physics
Arizona State University
email: oliver.beckstein@asu.edu
web: http://becksteinlab.physics.asu.edu

**Comment by orbeckst** [comment]
Need to fix #30 for the tests to make sense.

**Comment by orbeckst** [comment]
Apparently it does not work to decorate fixtures with xfail so I added the decorator to the tests and amended the history.

File: pmda/test/test_rms.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    def correct_values_frame_5(self):
+        return [[5, 6.0, 0.91544906]]
+
+    def test_rmsd(self, universe):
+        ca = universe.select_atoms("name CA")
+        universe.trajectory.rewind()
+        RMSD = rms.RMSD(ca, ca).run(n_jobs=2)
+        assert_array_equal(RMSD.rmsd.shape, (universe.trajectory.n_frames, 3))
+
+    @xfail_mda_issue1819
```
Comment (orbeckst) [inline pmda/test/test_rms.py]: Locally the xfail works (eg when I change 0.17.0 to 0.19.0) but not on travis. Not sure why.

File: pmda/test/test_rms.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+from distutils.version import LooseVersion
+
+import MDAnalysis
+from MDAnalysisTests.datafiles import (PSF, DCD)
+
+from pmda import rms
+
+# remove when MDAnalysis 0.18.0 is available
+xfail_mda_issue1819 = pytest.mark.xfail(
+    LooseVersion(MDAnalysis.__version__) <= LooseVersion("0.17.0"),
```
Comment (orbeckst) [inline pmda/test/test_rms.py]: Local testing gives proper xfails when I set `LooseVersion("0.19.0")`.

**Comment by orbeckst** [comment]
I don't understand why travis still fails, even with the xfails in place.

**Comment by orbeckst** [comment]
When I try the test locally in a conda env with MDA 0.17.0 I get
```
(pmda) yngvi:pmda oliver$ pytest -v pmda/test/test_rms.py
============================= test session starts ==============================
platform darwin -- Python 3.6.5, pytest-3.5.0, py-1.5.3, pluggy-0.6.0 -- /Users/oliver/anaconda3/envs/pmda/bin/python
cachedir: .pytest_cache
rootdir: /Users/oliver/MDAnalysis/pmda, inifile:
plugins: xdist-1.22.2, forked-0.2
collected 3 items

pmda/test/test_rms.py::TestRMSD::test_rmsd PASSED                        [ 33%]
pmda/test/test_rms.py::TestRMSD::test_rmsd_step xfail                    [ 66%]
pmda/test/test_rms.py::TestRMSD::test_rmsd_single_frame xfail            [100%]

===================== 1 passed, 2 xfailed in 1.66 seconds ======================
```

so the xfail works locally.

**Comment by orbeckst** [comment]
I am restarting the build ... just because.

**Comment by orbeckst** [comment]
@kain88-de any idea why the xfail decorator is not working on travis? https://travis-ci.org/MDAnalysis/pmda/jobs/364000696#L2064

**Comment by kain88-de** [comment]
did you compare pytest versions

**Comment by orbeckst** [comment]
Same versions:

- on travis Py 3.5: pytest-3.5.0
- Locally: pytest-3.5.0

**Comment by orbeckst** [comment]
I think I understand the problem: Travis uses the MDA 0.17.1-dev from @kain88-de 's channel and I don't think that [those builds](https://anaconda.org/kain88-de/mdanalysis) already have the fix for MDAnalysis/mdanalysis#1819 (because they were updated last time 2 months ago). BUT: these builds are versioned 0.17.1-dev and so the xfail considers them tests that *should* pass.

My local tests passed with MDAnalysis 0.17.0 and the xfail behaved properly. Should I switch pmda tests to using the the standard MDA distribution or is there another reason that we are using the develop builds?

**Comment by kain88-de** [comment]
tests pass now. I also merged now that we test against the stable version of mdanalysis

**Pull request merged**